src/main/kotlin/de/bund/digitalservice/useid/events/EventService.kt

```diff
@@ -35,8 +41,16 @@ class EventService {
      */
     fun publish(event: ServerSentEvent<Any>, widgetSessionId: UUID) {
         log.info { "Publish event to consumer $widgetSessionId" }
-        val consumer = consumers[widgetSessionId] ?: throw ConsumerNotFoundException(widgetSessionId)
-        consumer.accept(event)
+        val emitter = consumers[widgetSessionId] ?: throw ConsumerNotFoundException(widgetSessionId)
+
+        nonBlockingService.execute {
+            try {
+                emitter.send(event)
+                emitter.complete()
```

Comment:
I'm not sure, we need this call after each `send`. It's not a blocker, since this is prototype code, I'll see when I can verify that and have it on my list.